src/megaapi_impl.cpp

```diff
@@ -23166,96 +23166,118 @@ void MegaApiImpl::sendPendingRequests()
+            // generate fresh random key for this folder node
+            byte buf[FOLDERNODEKEYLENGTH];
+            client->rng.genblock(buf, FOLDERNODEKEYLENGTH);
+            newnode->nodekey.assign((char*)buf, FOLDERNODEKEYLENGTH);
+            SymmCipher key;
+            key.setkey(buf);
 
-                fireOnSyncAdded(sync.get(), MegaSync::NEW);
-                fireOnRequestFinish(request, make_unique<MegaErrorPrivate>(e));
+            // generate fresh attribute object with the folder name
+            AttrMap attrs;
+            fsAccess->normalize(&backupName);
+            attrs.map['n'] = backupName;
+            nameid attrId = AttrMap::string2nameid("dev-id"); // "device-id" would be too long
+            attrs.map[attrId] = deviceId;
 
-                // Update deviceNameNode somehow?
-                // HOW ?
+            // JSON-encode object and encrypt attribute string
+            string attrstring;
+            attrs.getjson(&attrstring);
+            newnode->attrstring.reset(new string);
+            client->makeattr(&key, newnode->attrstring, attrstring.c_str());
 
-                // tag the new node with `dev:<device-id>` in `attrstring`
-                // HOW ?
-            }
+            // add the newly generated folder node
+            client->putnodes(deviceNameNode->nodehandle, move(newnodes));
 
-            // the name of the backup folder was set when the request was created
-            const char* backupName = request->getName();
+            // Get the new node & handle.
+            // HOW ? How can it wait until the remote folder was created?
+            handle bkpHandle = UNDEF;
 
-            // create "My Backups"/`DEVICE_NAME`/backupName/
-            // HOW ? Like this?
+            // create the SyncConfig
             auto nextSyncTag = client->nextSyncTag();
-            auto sync = make_unique<MegaSyncPrivate>(request->getFile(), request->getName(), deviceNameNode->getHandle(), nextSyncTag);
-            sync->setListener(request->getSyncListener());
+            std::unique_ptr<char[]> remotePath { getNodePathByNodeHandle(bkpHandle) };
+            SyncConfig syncConfig { nextSyncTag, nullptr, deviceName.c_str(), bkpHandle, remotePath.get(),
+                                    0, {}, true, SyncConfig::TYPE_BACKUP };
 
-            // NOW WHAT ? Something like this?
-            std::unique_ptr<char[]> remotePath{ getNodePathByNodeHandle(request->getNodeHandle()) };
-            SyncConfig syncConfig{ nextSyncTag, request->getFile(), request->getName(), deviceNameNode->getHandle(), remotePath.get(),
-                                   0, {}, true, SyncConfig::TYPE_BACKUP };
-
-            SyncError syncError = NO_SYNC_ERROR;
-            e = client->addsync(syncConfig, DEBRISFOLDER, nullptr, syncError, true, sync.get());
+            // What should be done with syncConfig ?  Something like this?
```

Comment:
In order to add the new sync for the backup folder, you can use the `case MegaRequest::TYPE_ADD_SYNC:` as reference. Basically, it needs to do the same. Again, feel free to reuse the code by encapsulating common parts in new functions, called from both places